x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -215,8 +216,19 @@ private static Failure fail(Node<?> source, String message, Object... args) {
         // if there are no (major) unresolved failures, do more in-depth analysis
 
         if (failures.isEmpty()) {
+            Set<Failure> localFailures = new LinkedHashSet<>();
             final Map<Attribute, Expression> collectRefs = new LinkedHashMap<>();
 
+            // Full-Text search function are not allowed in the SELECT clause
+            plan.forEachUp(p -> {
```

Comment:
Despite being a small snippet, can you please extract it into its own static method - it's easier to reason about its purpose before looking at its details?
Thanks.